PR#0001: Реализован паттерн Адаптер на примере цифровых и аналоговых часов

albrettto opened the pull request.

albrettto committed:
feat(analog_clock): Добавлен метод для получения текущего времени суток (день / ночь)

albrettto committed:
docs(analog_clock): Добавлена документация для аналоговых часов

albrettto committed:
feat(clock_adapter): Реализован адаптер аналоговых часов для цифровых часов

albrettto committed:
feat(main): Добавлен main файл

albrettto committed:
test(test): Добавлен файл с тестами

albrettto committed:
devops(CI): Добавлен файл сборки и тестирования

albrettto committed:
chore(files): Форматирование кода

vladushka2000 commented on clocks/analog_clock.py:
```diff
@@ -4,16 +4,35 @@
 
 class AnalogClock(BaseAnalogClock):
```
пишем докстрингу абсолютно для всех классов и методов

vladushka2000 commented on clocks/clock_adapter.py:
```diff
@@ -0,0 +1,131 @@
+
+    """
+
+    _HOURS = 12
```
вынести в отдельный класс с константами

vladushka2000 commented on clocks/clock_adapter.py:
```diff
@@ -0,0 +1,131 @@
+    _HOURS = 12
+    _MIN_OR_SEC = 60
+    _DEG_FOR_HOUR = 30
+    _DEG_FOR_MIN_OR_SEC = 6
```
пусть лучше будут 2 отдельные переменные, без OR

vladushka2000 commented on clocks/clock_adapter.py:
```diff
@@ -0,0 +1,131 @@
+        Returns:
+            float: Угол в градусах для часовой стрелки.
+        """
+        return (
```
сложновато воспринимается. Го промежуточные результаты записывать и потом возвращать уже итог. Это же касается _minute_to_angle()

vladushka2000 commented:
вынести логику преобразования времени/дат в отдельный статический класс хелпер

albrettto committed:
feat(time_consts): Добавлен класс с константами, используемыми для конвертации времени

albrettto committed:
feat(time_converter): Добавлен класс-помощник для преобразования времени

albrettto committed:
docs(files): Дописана недостающая документация

vladushka2000 commented on clocks/time_converter.py:
```diff
@@ -0,0 +1,82 @@
+            int: Число часов (от 0 до 23).
+        """
+        hour = int(hour_angle / TimeConsts.DEG_FOR_HOUR)
+        if day_night_division == DayNightDivision.PM and hour != 12:
```
что такое 12?) Не от всех магических чисел избавился

albrettto replied:
Я про это число помнил. Просто подумал, что в этом случае можно оставить и в виде числа

albrettto replied:
Исправил

albrettto committed:
chore(magic numbers): Убраны "магические" числа

vladushka2000 commented:
зачел

Rognak commented on tests/test.py:
```diff
@@ -0,0 +1,142 @@
+import unittest
```
Название файла не отражает того, что подвергается тестированию.

albrettto replied:
Изменил название файла

albrettto committed:
refactor(test): Изменено наименование файла с тестами

Наименование test.py было изменено на test_clock_adapter_functionality.py. Это было сделано для того, чтобы наименование файла отражало суть тестов

Rognak commented on tests/test_clock_adapter_functionality.py:
слово functionality здесь избыточно

Rognak reviewed (approved):
Исправь название файла

albrettto committed:
chore(test): Изменено наименование файла с тестами

Наименование test_clock_adapter_functionality.py было изменено на test_clock_adapter.py.
